### PR TITLE
adds release cycle schedule yaml to be used by schedule-builder

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

Description: With the release cycle support to schedule-builder (https://github.com/kubernetes/release/pull/2215), we are adding the required yaml template (which currently follows the 1.23 schedule) for the schedule-builder to work upon

### CODE REVIEW

**Comment by saschagrunert** [comment]
Thank you for taking the lead over this @mritunjaysharma394! Let's also add the generated output to this PR. It should link from there: https://kubernetes.io/releases/#upcoming-release (move the upcoming releases on top of the release history).

We can add the generated doc to `k8s.io/website/content/en/releases`, where it makes sense to add only the current release cycle.

File: data/releases/release-cycle-schedule.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      when: Wed December 15, 2021
+      week: week 17
+      ciSignal:
+      tldr: true
+    - what: Release retrospective part 3 (10:00am PST)
+      who: Community
+      when: Thu December 16, 2021
+      week: week 17
+      ciSignal:
+      tldr: true
```
Comment (saschagrunert) [inline data/releases/release-cycle-schedule.yaml:188]: Please add a newline at the end of the file :upside_down_face:

File: data/releases/release-cycle-schedule.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
+releases:
```
Comment (saschagrunert) [inline data/releases/release-cycle-schedule.yaml:1]: We should somewhere maintain documentation how to generate the schedule from this doc. How about adding a comment on top of this file?

File: data/releases/release-cycle-schedule.yaml (hunk 1 above)
Comment (mritunjaysharma394) [reply data/releases/release-cycle-schedule.yaml]: Thanks for the review, @saschagrunert , fixed it :smile:

File: data/releases/release-cycle-schedule.yaml (hunk 2 above)
Comment (mritunjaysharma394) [reply data/releases/release-cycle-schedule.yaml:21]: Thank you so much, have tried to add one, thanks a lot!

**Comment by mritunjaysharma394** [comment]
> Thank you for taking the lead over this @mritunjaysharma394! Let's also add the generated output to this PR. It should link from there: https://kubernetes.io/releases/#upcoming-release (move the upcoming releases on top of the release history).
> 
> We can add the generated doc to `k8s.io/website/content/en/releases`, where it makes sense to add only the current release cycle.
> 
> Let's manually compare the output of the generated schedule with the [current version](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.23) to ensure that we do not drop any information.

Thanks a lot for the review @saschagrunert! I have tried to implement your suggestions and have tried to check if the generated schedule matches with the current version or not but please do review and let me know, we fill fix it, thanks a lot!

File: content/en/releases/_index.md [hunk 3]
```diff
@@ -22,6 +22,6 @@ More information in the [version skew policy](/releases/version-skew-policy/) do
 
 ## Upcoming Release
 
-Check out the [schedule](https://github.com/kubernetes/sig-release/tree/master/releases/release-{{< skew nextMinorVersion >}}) for the upcoming **{{< skew nextMinorVersion >}}** Kubernetes release!
+Check out the [schedule](/release/upcoming-release-cycle-schedule.md) for the upcoming **{{< skew nextMinorVersion >}}** Kubernetes release!
```
Comment (saschagrunert) [inline content/en/releases/_index.md:25]: Let's move this section before the "Release History" above.

File: content/en/releases/upcoming-release-cycle-schedule.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
+---
```
Comment (saschagrunert) [inline content/en/releases/upcoming-release-cycle-schedule.md:1]: Let's rename that file to `release-cycle-schedule.md`.

File: content/en/releases/upcoming-release-cycle-schedule.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+title: Kubernetes Release Cycle Schedule
```
Comment (saschagrunert) [inline content/en/releases/upcoming-release-cycle-schedule.md:2]:
```suggestion
title: Release Cycle Schedule
```

File: content/en/releases/upcoming-release-cycle-schedule.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+## TL;DR
+
+The 1.23 release cycle is proposed as follows:
+
+- **Mon August 23, 2021**: week 1 - Release cycle begins
+- **Thu September 2, 2021**: week 2 - [Production Readiness Soft Freeze](https://groups.google.com/g/kubernetes-sig-architecture/c/a6_y81N49aQ)
+- **Thu September 9, 2021**: week 3 - [Enhancements Freeze](../release_phases.md#enhancements-freeze)
+- **Tue November 16, 2021**: week 13 - [Code Freeze](../release_phases.md#code-freeze)
+- **Tue November 23, 2021**: week 14 - [Test Freeze](../release_phases.md#test-freeze)
```
Comment (saschagrunert) [inline content/en/releases/upcoming-release-cycle-schedule.md:42]: Those lead into a 404, too.

File: content/en/releases/upcoming-release-cycle-schedule.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+[k8s1.23-calendar]: https://bit.ly/k8s-release-cal
+[Internal Contact Info]: https://bit.ly/k8s123-contacts
+[Retrospective Document]: https://bit.ly/k8s123-retro
+
+[Enhancements Freeze]: ../release_phases.md#enhancements-freeze
+[Burndown]: ../release_phases.md#burndown
+[Code Freeze]: ../release_phases.md#code-freeze
+[Exception]: ../release_phases.md#exceptions
+[Thaw]: ../release_phases.md#thaw
+[Test Freeze]: ../release_phases.md#test-freeze
```
Comment (saschagrunert) [inline content/en/releases/upcoming-release-cycle-schedule.md:102]: Those links are broken, too

File: content/en/releases/upcoming-release-cycle-schedule.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+[release-team@]: https://groups.google.com/a/kubernetes.io/g/release-team
+[kubernetes-sig-release@]: https://groups.google.com/forum/#!forum/kubernetes-sig-release
+[#sig-release]: https://kubernetes.slack.com/messages/sig-release/
+[kubernetes-release-calendar]: https://bit.ly/k8s-release-cal
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[master-blocking]: https://testgrid.k8s.io/sig-release-master-blocking#Summary
+[master-informing]: https://testgrid.k8s.io/sig-release-master-informing#Summary
+[1.23-blocking]: https://testgrid.k8s.io/sig-release-1.23-blocking#Summary
+[exception requests]: ../EXCEPTIONS.md
+[release phases document]: ../release_phases.md
```
Comment (saschagrunert) [inline content/en/releases/upcoming-release-cycle-schedule.md:112]: Another two broken links.

**Comment by reylejano** [comment]
/assign
/hold 

feel free to remove the hold after comments have been addressed

**Comment by jimangel** [comment]
@mritunjaysharma394 thanks for the PR! [k8s.io/releases](https://kubernetes.io/releases/) was created the intention was to capture all of the current releases (to the latest patch) and provide resources linked from that. The section for "Check out the schedule for the upcoming 1.23 Kubernetes release!" was added later for folks more interested in contributing / keeping up with the latest efforts.

The way this PR (#29569) is set up, it duplicates the source of the release schedule which changes very rapidly. I would prefer to single source the release schedule.

One alternative option might be to link directly to https://www.kubernetes.dev/resources/release/ for that sub section. That page is scraped from github.com at the end of the day (same location that the tool generates).

TL;DR: The schedule builder is great for generating the first-run of upcoming release dates, but given how frequently it changes, I don't know if the final generated one should land in k/website.

**Comment by reylejano** [comment]
> One alternative option might be to link directly to https://www.kubernetes.dev/resources/release/ for that sub section. That page is scraped from github.com at the end of the day (same location that the tool generates).

+1 to @jimangel 's suggestion to link directly to https://www.kubernetes.dev/resources/release/, as this page only needs one change at the beginning of a release cycle to bump [k/contributor-site/external-sources/kubernetes/sig-release](https://github.com/kubernetes/contributor-site/blob/master/external-sources/kubernetes/sig-release) to the current release cycle

**Comment by mritunjaysharma394** [comment]
Hi @saschagrunert , @jimangel and @reylejano , thanks a lot for the review and I am extremely sorry for getting back to this so late, I was actually on break from everything due to some personal life problems and my day-job commitments. 

I also agree to @jimangel 's suggestions and I think it already has been implemented since that page is scraped from github.com, so can we close this PR? 

Thanks a lot again :smile:

**Comment by reylejano** [comment]
Thank you @mritunjaysharma394, closing this PR
We should link to https://www.kubernetes.dev/resources/release/  as mentioned in the comments instead
/close

File: content/en/releases/upcoming-release-cycle-schedule.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+auto_generated: true
+---
+
+<!-- THIS CONTENT IS AUTO-GENERATED via schedule-builder in k/release -->
+
+# Kubernetes 1.23
+
+#### Links
+
+* [This document](https://git.k8s.io/sig-release/releases/release-1.23/README.md)
```
[the remark this answers is not in the document]
Comment (mritunjaysharma394) [reply content/en/releases/upcoming-release-cycle-schedule.md:13]: Hi @saschagrunert, I guess this is still relevant for the schedule-builder in k8s/release? Should I create a PR to remove the broken links or may be add the correct ones?
